python_modules/dagster/dagster/_utils/cached_method.py

```diff
@@ -62,14 +63,26 @@ def a_method(self, arg1, arg2):
     """
     cache_attr_name = method.__name__ + CACHED_METHOD_FIELD_SUFFIX
 
+    # Cache these once self is first observed to avoid expensive work on each access
     arg_names = None
+    is_dagster_model = None
```

Comment:
using this script https://gist.github.com/alangenfeld/766570ee4f29229af1c2ec774f20ee1c

master: 
```
Only kwargs cached: 7.51351750001777
Mixed kwargs cached: 8.863711125100963
Only ordinals cached: 8.780334583017975
```

before this optimization:
```
Only kwargs cached: 10.084340583067387
Mixed kwargs cached: 12.293588166008703
Only ordinals cached: 11.324828165932558
```

after
```
Only kwargs cached: 7.32807208399754
Mixed kwargs cached: 9.117478957981803
Only ordinals cached: 8.596969916019589
```